### DISCUSSION TITLE
Insert template contents at the start of the file?

### DISCUSSION THREAD

**Discussion by balaji-dutt** [SilentVoid13/Templater#307] [Help]
Hi,

I've created a Templater that inserts a bunch of YAML frontmatter so the files can be used in a Hugo blog. However, I do not want to apply the template to all files by default so this is a template that will only be run _after_ I've finished writing the note.

Since the YAML frontmatter needs to be at the top of the file, I have to remember to add a newline at the start of the file and position my cursor at the start before applying the template. Is there a way for me to have the template be automatically inserted at the start of the file when I apply the template, rather than having to ensure my cursor is at the start of the file?

**Comment by AB1908** [comment]
You'd have to use the Obsidian API for this sadly.

**Reply by balaji-dutt** [reply]
Thanks for the clarification @AB1908. I'll try to ask on the Obsidian forums and see if folks can help me figure out what I need to do with the Obsidian API to try and implement this.

**Reply by AB1908** [reply]
`await app.vault.modify(TFile, "contents")`

Contents should be your stuff prepended to the existing content.

**Reply by liamcain** [reply]
`modify` doesn't prepend, it replaces the entire contents of the file. You're better off using `replaceRange`. 

Something like this would prepend to the current file:

```
<%* window.app.workspace.activeLeaf.view.editor.replaceRange("---\n\n---\n", { line: 0, ch: 0 }, { line: 0, ch: 0 }) %>
```

**Reply by AB1908** [reply]
That's definitely a better way haha. I never did learn much about the workspace related stuff.

**Reply by kurakurakantor** [reply]
@liamcain thank you so much for this!

**Reply by kurakurakantor** [reply]
also for people like me who were wondering how to make cursor jump works with it, you need to add focus after that, here's an example:
```
<%* window.app.workspace.activeLeaf.view.editor.replaceRange("[[<% tp.file.cursor(1) %>]]", { line: 0, ch: 0 }, { line: 0, ch: 0 }) -%>
<%* window.app.workspace.activeLeaf.view.editor.focus() %>
```

**Reply by AB1908** [reply]
Oh thanks! TIL! I haven't done much `workspace` stuff yet haha.

**Reply by balaji-dutt** [reply]
Holy thread necro Batman! Sorry @liamcain that it's taken me *this long* to respond but yes `replaceRange` did do the trick! I struggled a little bit as I needed to replace the entire text at the start of my note with the new Template contents but after much trial and error I finally wound up with this unholy mess:

~~~
<%* endch = window.app.workspace.activeLeaf.view.editor.getLine(0).length %>
<%* window.app.workspace.activeLeaf.view.editor.replaceRange('', { line: 0, ch: 0 },{ line: 0, ch: endch }) -%>
<%* window.app.workspace.activeLeaf.view.editor.focus() %>
~~~
